File: ListaDuplamenteEncad/ListaDuplamenteEncad.c
#include <stdio.h>
#include <stdlib.h>

typedef struct lista {
    struct lista *anterior;
    int posicao;
    int elemento;
    struct lista *proximo;
} Lista;

int main () {
    Lista *ponteiroInicial;
    Lista *ponteiroAtual;
    int resp;

    ponteiroInicial = (Lista *)malloc(sizeof(Lista));
    ponteiroInicial->anterior = NULL;
    ponteiroAtual = ponteiroInicial;

    do {
        printf("Entre com a posicao do elemento: ");
        scanf("%d", &ponteiroAtual->posicao);

        printf("Entre com o valor da posicao [%d]: ", ponteiroAtual->posicao);
        scanf("%d", &ponteiroAtual->elemento);

        printf("\nDeseja continuar? \n\n[1] - Sim\n[2] - Nao\n\nSua Resposta: ");
        scanf("%d", &resp);
        system("cls");

        if (resp == 1) {
            ponteiroAtual->proximo = (Lista *)malloc(sizeof(Lista));
            ponteiroAtual->proximo->anterior = ponteiroAtual->posicao;
            ponteiroAtual = ponteiroAtual->proximo;
        } else {
            break;
        }

    } while (resp != 2);

    ponteiroAtual->proximo = NULL;

    while (ponteiroInicial != NULL) {
        printf("Posicao Anterior: %d\n", ponteiroInicial->anterior);
        printf("POSICAO ATUAL: %d\n", ponteiroInicial->posicao);
        printf("ELEMENTO SALVO: %d\n", ponteiroInicial->elemento);
        
        if (ponteiroInicial->proximo == NULL) {
            printf("Posicao Proximo: %d\n", 0);
        } else {
            printf("Posicao Proximo: %d\n", ponteiroInicial->proximo->posicao);
        }
            
        printf("\n");
        ponteiroInicial = ponteiroInicial->proximo;
    }

    system("pause");
    return 0;
}

File: ListaDuplamenteEncad/ListaDuplamenteEncad.py
class ListaEncad:
    def __init__ (self):
        self.anterior = None
        self.posicao = 0
        self.elemento = 0
        self.proximo = None

ponteiroInicial = ListaEncad()
ponteiroAtual = ponteiroInicial

while True:
    print("-------------- ENTRADA --------------\n")
    ponteiroAtual.posicao = int(input("Entre com a posicao do Elemento: "))
    ponteiroAtual.elemento = int(input("Entre com o valor da posicao [{}]: ".format(ponteiroAtual.posicao)))
    resp = int(input("\nDeseja Continuar?\n\n[1] - Sim\n[2] - Nao\n\nSUA ESCOLHA: "))
    print()

    if resp == 1:
        ponteiroAtual.proximo = ListaEncad()
        ponteiroAtual.proximo.anterior = ponteiroAtual.posicao
        ponteiroAtual = ponteiroAtual.proximo
    else:
        break

while ponteiroInicial is not None:
    print("Posicao Anterior: ", ponteiroInicial.anterior)
    print("POSICAO ATUAL: ", ponteiroInicial.posicao)
    print("ELEMENTO SALVO: ", ponteiroInicial.elemento)

    if ponteiroInicial.proximo is None:
        print("Posicao Proximo: None")
    else:
        print("Posicao Proximo: ", ponteiroInicial.proximo.posicao)

    print()
    ponteiroInicial = ponteiroInicial.proximo

